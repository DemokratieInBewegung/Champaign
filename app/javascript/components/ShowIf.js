// @flow
import React from 'react';
import type { Element } from 'react';
import classnames from 'classnames';
import './ShowIf.css';

// <ShowIf /> *renders* its children, but hides them with CSS if
// `props.condition` is falsy, as opposed to not rendering the node at all
// which won't trigger the lifecycle methods. Use this when you want the component
// to be mounted but not displayed, and its lifecycle methods to be triggered,
// but you don't want to display the component, and you don't want to add that
// show/hide display logic to the child components.
type OwnProps = {
  condition: boolean,
  children?: Element<any>,
};
export default (props: OwnProps) => {
  if (!props.children) return null;

  const classNames: string = classnames({
    ShowIf: true,
    'ShowIf--visible': props.condition,
    'ShowIf--hidden': !props.condition,
  });

  return (
    <div className={classNames}>
      {props.children}
    </div>
  );
};
