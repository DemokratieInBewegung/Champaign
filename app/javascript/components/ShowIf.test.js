// @flow
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ShowIf from './ShowIf';

test('renders correctly', () => {
  const wrapper = shallow(<ShowIf condition={true}>Content</ShowIf>);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test('renders null if passed with no children', () => {
  const wrapper = shallow(<ShowIf condition={true} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test('it has visible class when condition is truthy', () => {
  const wrapper = shallow(<ShowIf condition={true}>Hello</ShowIf>);
  expect(wrapper.hasClass('ShowIf--visible')).toBeTruthy();
});

test('it has hidden class when condition is falsy', () => {
  const wrapper = shallow(<ShowIf condition={false}>Hello</ShowIf>);
  expect(wrapper.hasClass('ShowIf--hidden')).toBeTruthy();
});
