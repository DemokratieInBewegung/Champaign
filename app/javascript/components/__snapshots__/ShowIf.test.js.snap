// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ShowIf ShowIf--visible"
>
  Content
</div>
`;

exports[`renders null if passed with no children 1`] = `null`;
